### PR TITLE
[client] add SignatureType supporting classes (#11)

### PR DESCRIPTION
**Pull request by antoinemzs**
### Proposed changes

* New classes for standardising the creation of signature types structures in collectors

### Related issues

* Contributes https://github.com/OpenBAS-Platform/collectors/issues/11

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: pyobas/signatures/types.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class MatchTypes:
+    MATCH_TYPE_FUZZY = "fuzzy"
```
Comment (savacano28) [inline pyobas/signatures/types.py:2]: Maybe declare it as an Enum.

File: pyobas/signatures/signature_match.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from pyobas.exceptions import OpenBASError
+from pyobas.signatures.types import MatchTypes
+
+
+class SignatureMatch:
+    def __init__(self, match_type: str, match_score: int):
+        if not match_score and match_type != MatchTypes.MATCH_TYPE_SIMPLE:
+            raise OpenBASError(
```
Comment (savacano28) [inline pyobas/signatures/signature_match.py:8]: Maybe use: if match_score is None ? what happens if match_score is 0? not 0 === not None => true, non? match_score could be 0?

File: test/signatures/test_signature_match.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        score = None
+        self.assertRaises(
+            OpenBASError, SignatureMatch, MatchTypes.MATCH_TYPE_FUZZY, score
+        )
+
+    def test_simple_match_with_null_score_does_not_throw(self):
+        score = None
+        SignatureMatch(match_type=MatchTypes.MATCH_TYPE_SIMPLE, match_score=score)
+
+
```
Comment (savacano28) [inline test/signatures/test_signature_match.py:19]: Add test score= 0

File: pyobas/signatures/signature_match.py (hunk 2 above)
Comment (antoinemzs) [reply pyobas/signatures/signature_match.py:8]: Yes good spot

File: pyobas/signatures/signature_type.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        self,
+        label: str,
+        match_type: str = MatchTypes.MATCH_TYPE_SIMPLE,
+        match_score: int = None,
+    ):
+        self.label = label
+        self.match_policy = SignatureMatch(match_type, match_score)
+
+    def make_struct_for_matching(self, data):
+        struct = {
```
Comment (savacano28) [inline pyobas/signatures/signature_type.py:16]: Maybe we could need some doc?

File: .circleci/config.yml [hunk 5]
```diff
@@ -32,6 +32,21 @@ jobs:
       - ms-teams/report:
           only_on_fail: true
           webhook_url: $MS_TEAMS_WEBHOOK_URL
+  test:
+    docker:
```
Comment (savacano28) [inline .circleci/config.yml:36]: Thanks!

**Comment by antoinemzs** [comment]
@savacano28 hopefully the latest commits resolve your comments 🙏

**Review by savacano28 — APPROVED** [review]

File: pyobas/__init__.py [hunk 6]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-__version__ = "1.9.0"
+__version__ = "1.10.0"
```
[the remark this answers is not in the document]
Comment (antoinemzs) [reply pyobas/__init__.py:2]: since we're on the 1.10.0 branch, does it not make sense?

**Pull request merged**